From 8dd36c42cf841fa4f4d97515ab77b327ecd5c1c7 Mon Sep 17 00:00:00 2001
From: Liu Shixin <liushixin2@huawei.com>
Date: Mon, 23 May 2022 21:18:20 +0800
Subject: [PATCH] dynamic_hugetlb: fix undefined struct cftype
MIME-Version: 1.0
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 8dd36c42cf841fa4f4d97515ab77b327ecd5c1c7
Modified-by-SEL: No

Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

hulk inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I576NI
CVE: NA
backport: openEuler-22.03-LTS

--------------------------------

There is a build warning when !CONFIG_CGROUPS:

In file included from ./include/linux/memcontrol.h:25,
                 from ./include/linux/swap.h:10,
                 from ./include/linux/suspend.h:5,
                 from drivers/cpuidle/cpuidle.c:23:
./include/linux/dynamic_hugetlb.h:115:47: warning: ‘struct cftype’ declared
inside parameter list will not be visible outside of this definition or
declaration
  115 | static inline bool dhugetlb_hide_files(struct cftype *cft)
      |                                               ^~~~~~

Since the function is only invoked when CONFIG_CGROUPS enabled, we can fix it
by restricting its definition to on CONFIG_CGROUPS.

Signed-off-by: Liu Shixin <liushixin2@huawei.com>
Reviewed-by: Kefeng Wang <wangkefeng.wang@huawei.com>
Reviewed-by: Hanjun Guo <guohanjun@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 include/linux/dynamic_hugetlb.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/linux/dynamic_hugetlb.h b/include/linux/dynamic_hugetlb.h
index 237a7329ff64..5dcba8e8b933 100644
--- a/include/linux/dynamic_hugetlb.h
+++ b/include/linux/dynamic_hugetlb.h
@@ -112,10 +112,12 @@ void free_huge_page_to_dhugetlb_pool(struct page *page, bool restore_reserve);
 
 struct dhugetlb_pool {};
 
+#ifdef CONFIG_CGROUPS
 static inline bool dhugetlb_hide_files(struct cftype *cft)
 {
 	return false;
 }
+#endif
 static inline void hugetlb_pool_inherit(struct mem_cgroup *memcg, struct mem_cgroup *parent)
 {
 }
-- 
2.34.1

